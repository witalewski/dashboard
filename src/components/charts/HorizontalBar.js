import React from 'react';
import styled from '@emotion/styled';
import {
  chartBackground,
  chartColors,
  textColor,
} from '../../global/styleConstants';

const StyledHorizontalBar = styled.div`
  position: relative;

  width: 100%;
  height: 32px;

  margin: 8px;

  background: ${chartBackground};

  .bar {
    height: 100%;
    background: ${chartColors[0]};
  }

  .label {
    position: absolute;
    top: 50%;
    font-size: 12px;
    color: white;
    transform: translate(calc(-100% - 12px), -50%);
  }

  .label--outside {
    color: ${textColor};
    transform: translate(8px, -50%);
  }
`;

export const HorizontalBar = ({ percent }) => (
  <StyledHorizontalBar>
    <div className="bar" style={{ width: `${percent}%` }} />
    <div
      className={`label ${percent < 5 && 'label--outside'}`}
      style={{ left: `${percent}%` }}
    >
      {percent}%
    </div>
  </StyledHorizontalBar>
);
