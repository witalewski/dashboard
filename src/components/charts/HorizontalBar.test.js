import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { HorizontalBar } from './HorizontalBar';

describe('HorizontalBar', () => {
  it('matches snapshot at 0%', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<HorizontalBar percent={0} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
  it('matches snapshot at 2%', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<HorizontalBar percent={2} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
  it('matches snapshot at 5%', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<HorizontalBar percent={5} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
  it('matches snapshot at 10%', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<HorizontalBar percent={10} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
  it('matches snapshot at 50%', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<HorizontalBar percent={50} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
  it('matches snapshot at 75%', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<HorizontalBar percent={75} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
  it('matches snapshot at 100%', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<HorizontalBar percent={100} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
