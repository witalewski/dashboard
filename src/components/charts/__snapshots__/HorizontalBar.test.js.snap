// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HorizontalBar matches snapshot at 0% 1`] = `
<ForwardRef(render)>
  <div
    className="bar"
    style={
      Object {
        "width": "0%",
      }
    }
  />
  <div
    className="label label--outside"
    style={
      Object {
        "left": "0%",
      }
    }
  >
    0
    %
  </div>
</ForwardRef(render)>
`;

exports[`HorizontalBar matches snapshot at 2% 1`] = `
<ForwardRef(render)>
  <div
    className="bar"
    style={
      Object {
        "width": "2%",
      }
    }
  />
  <div
    className="label label--outside"
    style={
      Object {
        "left": "2%",
      }
    }
  >
    2
    %
  </div>
</ForwardRef(render)>
`;

exports[`HorizontalBar matches snapshot at 5% 1`] = `
<ForwardRef(render)>
  <div
    className="bar"
    style={
      Object {
        "width": "5%",
      }
    }
  />
  <div
    className="label false"
    style={
      Object {
        "left": "5%",
      }
    }
  >
    5
    %
  </div>
</ForwardRef(render)>
`;

exports[`HorizontalBar matches snapshot at 10% 1`] = `
<ForwardRef(render)>
  <div
    className="bar"
    style={
      Object {
        "width": "10%",
      }
    }
  />
  <div
    className="label false"
    style={
      Object {
        "left": "10%",
      }
    }
  >
    10
    %
  </div>
</ForwardRef(render)>
`;

exports[`HorizontalBar matches snapshot at 50% 1`] = `
<ForwardRef(render)>
  <div
    className="bar"
    style={
      Object {
        "width": "50%",
      }
    }
  />
  <div
    className="label false"
    style={
      Object {
        "left": "50%",
      }
    }
  >
    50
    %
  </div>
</ForwardRef(render)>
`;

exports[`HorizontalBar matches snapshot at 75% 1`] = `
<ForwardRef(render)>
  <div
    className="bar"
    style={
      Object {
        "width": "75%",
      }
    }
  />
  <div
    className="label false"
    style={
      Object {
        "left": "75%",
      }
    }
  >
    75
    %
  </div>
</ForwardRef(render)>
`;

exports[`HorizontalBar matches snapshot at 100% 1`] = `
<ForwardRef(render)>
  <div
    className="bar"
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <div
    className="label false"
    style={
      Object {
        "left": "100%",
      }
    }
  >
    100
    %
  </div>
</ForwardRef(render)>
`;
